### DISCUSSION TITLE
Beginner question: File uploading Typescript error

### DISCUSSION THREAD

**Discussion by johannesbremer** [pocketbase/pocketbase#2626] [Q&A] [answered]
First of all I really love pocketbase and I would have never thought, that I could develop a website so quickly!
I know maybe this not the right place to ask about a typescript error, but it occurs when I just follow your documentation, so I thought I might give it a shot!
When I try to follow the documentation for uploading a file (https://pocketbase.io/docs/files-handling/#uploading-files)
I encounter the following TS error "Property 'files' does not exist on type 'HTMLElement'. (TS 2339)"
for the line "for (let file of fileInput.files){"
Many thanks in advance!

**Comment by ganigeorgiev** [comment, marked as answer]
You can try to explicitly set the type of the variable to [`HTMLInputElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement) OR type assert it like:
`(fileInput as HTMLInputElement).files`

**Reply by johannesbremer** [reply]
Thank you very much!
